tests/integration/servicecatalog/cmd_service_test.go

```diff
@@ -52,6 +52,25 @@ var _ = Describe("odo service command tests", func() {
 		})
 	})
 
+	Context("check search functionality", func() {
+
+		It("should pass with searching for part of a service name", func() {
+
+			// We just use "sql" as some catalogs only have postgresql-persistent and
+			// others dh-postgresql-db. So let's just see if there's "any" postgresql to begin
+			// with
+			output := helper.CmdShouldPass("odo", "catalog", "search", "service", "sql")
+			Expect(output).To(ContainSubstring("postgresql"))
+		})
+
+		It("should fail searching for something nonexistent", func() {
+
+			output := helper.CmdShoulFail("odo", "catalog", "search", "service", helper.RandString(20))
+			Expect(output).To(ContainSubstring("postgresql"))
```

Comment:
If it fails to search then IMO it should error out with a proper message of something like ```Service <service_name> not found in catalog list services```